modules/openthread/Kconfig

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config OPENTHREAD
```

Comment:
How are we going to choose between this implementation and the one in Zephyr? How does it work on this PR without having conflicts with the same Kconfig names from Zephyr?

Reply:
```
set(ZEPHYR_OPENTHREAD_CMAKE_DIR ${CMAKE_CURRENT_LIST_DIR}/openthread)
set(ZEPHYR_OPENTHREAD_KCONFIG ${CMAKE_CURRENT_LIST_DIR}/openthread/Kconfig)
```
in `modules.cmake` fully override cmake and kconfig in Zephyr `modules/openthread`, so these from upstream will always be ignored